### PR TITLE
Use slippy map projection fns that have been implemented in lonlat

### LINKED ISSUE #771: Always show analysis bounds on map
Currently we only show the bounds of the analysis on the map while "set custom geographic bounds" is active. We could just show the bounds all the time (with a non-filled dotted line rectangle or even just the corners). This way there would be no surprise effects when regional analyses were launched (and even on single-points people would visually see if their opportunity counts were being limited to a sub-area). When "bounds of analysis" was set to "same as existing analysis X" the bounds would update, and you would see the bounds rather than waiting until a regional analysis is already running.

### PR DESCRIPTION
**Pull request by trevorgerhardt**
Switch from Leaflet projection functions to Slippy map based functions that are the same used in R5 for coordinate conversion. Geographic used for bounds will snap to the same grid point as used on the backend. So this will affect:

- Display of all isochrones (they will be one "grid" pixel larger)
- Bounds of newly created regions 
- Custom bounds for analyses 

Depends on https://github.com/conveyal/lonlat/pull/18 and https://github.com/conveyal/jsolines/pull/39

closes #771

### CODE REVIEW

**Comment by ansoncfit** [comment]
I think there's an issue in the StopLayer coming from lonlat:

image

File: lib/components/modifications-map/transit-editor/gtfs-stop-gridlayer.js [hunk 1]
```diff
@@ -43,25 +44,27 @@ export default class GTFSStopGridLayer extends MapLayer {
...
-        s.stop_lon > topLeft.lng &&
-        s.stop_lon < botRight.lng
+        s.stop_lon > topLeft.lon &&
+        s.stop_lon < botRight.lon
       const drawStopToCanvas = s => {
         // get coordinates
-        // lat first for leaflet, every so often Lineland seems like a good idea
-        // http://www.gutenberg.org/ebooks/97
-        let {x, y} = map.project([s.stop_lat, s.stop_lon], z)
+        let {x, y} = lonlat.toPixel([s.stop_lat, s.stop_lon], z)
```
Comment (trevorgerhardt) [inline lib/components/modifications-map/transit-editor/gtfs-stop-gridlayer.js]: Whoops. @ansoncfit I removed the comment intending to switch to `[s.stop_lon, s.stop_lat]` but never did.

File: lib/components/directional-markers.js [hunk 2]
```diff
@@ -1,111 +1,137 @@
...
-        []
-      )
-    }
+function drawTile (
+  markers: LonLatBearing[],
+  ctx: CanvasRenderingContext2D,
+  tile: Point,
+  z: number
+) {
+  drawMarkersInTile(markers, tile, z, ({x, y}, bearing) => {
```
[the remark this answers is not in the document]
Comment (trevorgerhardt) [reply lib/components/directional-markers.js]: Hiding them isn't really a valid solution. An entire length of a line could be along an edge. 

We could revert back to doing them with markers, it really shouldn't be too slow and they appear on their own layer.

**Comment by trevorgerhardt** [comment]
The displayed bounds and displayed isochrones should be different with this PR. The current state from our previous change is that the bounds are snapped in incorrect places and the isochrones get displayed incorrectly smaller than it's bounds. 

This change displays the isochrones fully in it's bounds (bounds that are converted from geographic to grid pixel coordinates on the backend). But since bounds are saved in their geographic representation they will still be displayed incorrectly until re-positioned. New ones should look perfect.

File: lib/components/directional-markers.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ansoncfit) [reply lib/components/directional-markers.js]: I hadn't noticed it before, but yes, it is currently occurring.

**Pull request merged**